lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    AuthError,
+    Authority,
+    CacheManager,
+    ClientAuthErrorCodes,
+    Constants,
+    HeaderNames,
+    INetworkModule,
+    Logger,
+    NetworkRequestOptions,
+    ResponseHandler,
+    ServerAuthorizationTokenResponse,
+    TimeUtils,
+    createClientAuthError,
+} from "@azure/msal-common";
+import { ManagedIdentityId } from "../../config/ManagedIdentityId";
+import { ManagedIdentityRequestParameters } from "../../config/ManagedIdentityRequestParameters";
+import { CryptoProvider } from "../../crypto/CryptoProvider";
+import { ManagedIdentityRequest } from "../../request/ManagedIdentityRequest";
+import { HttpMethod } from "../../utils/Constants";
+import { ManagedIdentityResult } from "../../response/ManagedIdentityResult";
+import { ManagedIdentityUtils } from "../../utils/ManagedIdentityUtils";
+
+export abstract class BaseManagedIdentitySource {
+    protected logger: Logger;
+    private cacheManager: CacheManager;
+    private networkClient: INetworkModule;
```

Comment:
I couldn't find it but what is the actual implementation of INetworkMobile that gets passed in here?

Reply:
It's just a stub with two methods that aren't implemented: sendGetRequestAsync and sendPostRequestAsync. It's in msal-common: lib/msal-common/src/network/INetworkModule.ts